GitExtUtils/FileLogging/LogFileCleaner.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Threading;
+
+namespace GitExtUtils.FileLogging
+{
+    public class LogFileCleaner
+    {
+        private readonly LogFilenameCreator _logFilenameCreator;
+
+        public LogFileCleaner(LogFilenameCreator logFilenameCreator)
+        {
+            _logFilenameCreator = logFilenameCreator;
+        }
+
+        public void Run()
+        {
+            WaitToPreventAddedStartupTime();
+
+            var deleteThreshold = DateTime.Now - TimeSpan.FromDays(30);
```

Comment:
🤔 I think this may be way too long. 
A lot of dev (including myself) run multiple GE all day every day... how many logs will we be generating?
I think we shouldn't do more than 7, and realistically more than 2 days is probably isn't needed.

Reply:
It is a trade off between a bit of space waste compared to useful info in the logs. Eg. have the startup time increased significantly recently? how many git commands are fired every time window X is opened etc.  For those kinds of questions longer time lines are very convenient.

Also time time from you see something until you report a bug and until someone realises perhaps sending a log is a good idea and finding the time to reply that and the time for the issue raiser to read that response - could be more than 7 days ;)

I'd say more than 7 but we can do whatever and in a later PR make it configurable perhaps.